examples/specs_integration/src/sync/godot_sync_state.rs

```diff
@@ -0,0 +1,69 @@
+use godot::*;
+use specs::prelude::*;
+
+use crate::helper::try_load_scene;
+use singleton::try_get_singleton_state;
+use specs::world::Index;
+use specs::{RunNow, World};
+use std::collections::HashMap;
+use sync::{
+    GDDeleteOldSpatials, GdEntityCreation, GdEntityTransform, Template3dEntity, Template3dName,
+};
+
+pub struct GodotStateSync {
+    /// Please be aware that you should only use this class from the main Godot thread.
+    ///  Also note that if a spatial stored here is deleted by other code or scripts in your game
+    ///  then weird things will happen
+    templates: HashMap<Template3dName, Template3dEntity>,
+    spatials: HashMap<Index, Spatial>,
+    gd_deleter: Option<GDDeleteOldSpatials>,
+}
+
+impl GodotStateSync {
+    pub fn new() -> GodotStateSync {
+        GodotStateSync {
+            templates: Default::default(),
+            spatials: Default::default(),
+            gd_deleter: None,
+        }
+    }
+
+    pub fn do_sync(&mut self, owner: gdnative::Spatial, sync_func: fn(&mut World) -> ()) {
+        // Get the state if the mutex is not already locked right now...
+        let mut your_state = if let Some(state) = try_get_singleton_state() {
+            state
+        } else {
+            return;
+        };
+
+        let world = your_state.world_mut();
+
+        if self.gd_deleter.is_none() {
+            self.gd_deleter = Some(GDDeleteOldSpatials::new(world));
+        }

```

Comment:
[`Option::get_or_insert_with`](https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert_with) might be more communicative of the intention here.